plugin/forward/health_test.go

```diff
@@ -39,7 +44,43 @@ func TestHealth(t *testing.T) {
 	time.Sleep(1 * time.Second)
 	i1 := atomic.LoadUint32(&i)
 	if i1 != expected {
-		t.Errorf("Expected number of health checks to be %d, got %d", expected, i1)
+		t.Errorf("Expected number of health checks with RecursionDesired==true to be %d, got %d", expected, i1)
+	}
+}
+
+func TestHealthNoRecursion(t *testing.T) {
+	const expected = 1
```

Comment:
We only care that recursion is unset. The other tests takes care of counting. Please minimize what you're trying to test here.

Reply:
I can change the `expected` to a bool, but i still have to count. At least fail one query for the healthchecker to kick in.